packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -251,250 +191,29 @@ export class OdspDocumentService implements IDocumentService {
-            const props = {
-                entryTime: response.entryTime,
-                refreshSessionDurationSeconds:
-                    response.joinSessionResponse.refreshSessionDurationSeconds,
-                refreshAfterDeltaMs: response.refreshAfterDeltaMs,
-            };
-            if (response.refreshAfterDeltaMs > 0) {
-                this.scheduleJoinSessionRefresh(response.refreshAfterDeltaMs)
-                    .catch((error) => {
-                        const canRetry = canRetryOnError(error);
-                        // Only record error event in case it is non retriable.
-                        if (!canRetry) {
-                            this.mc.logger.sendErrorEvent({
-                                eventName: "JoinSessionRefreshError",
-                                details: JSON.stringify(props),
-                            },
-                            error,
-                            );
-                        }
-                    });
-            } else {
-                // Logging just for informational purposes to help with debugging as this is a new feature.
-                this.mc.logger.sendTelemetryEvent({
-                    eventName: "JoinSessionRefreshNotScheduled",
-                    details: JSON.stringify(props),
-                });
-            }
+    private async getDelayLoadedDocumentService() {
+        if (this.odspDocumentServiceDelayLoaded) {
+            return this.odspDocumentServiceDelayLoaded;
         }
-        return response.joinSessionResponse;
-    }
-
-    private calculateJoinSessionRefreshDelta(responseFetchTime: number, refreshSessionDurationSeconds: number) {
-        // 30 seconds is buffer time to refresh the session.
-        return responseFetchTime + ((refreshSessionDurationSeconds * 1000) - 30000) - Date.now();
-    }
-
-    /**
-     * Creats a connection to the given delta stream endpoint
-     *
-     * @param tenantId - the ID of the tenant
-     * @param documentId - document ID
-     * @param token - authorization token for delta service
-     * @param io - websocket library
-     * @param client - information about the client
-     * @param webSocketUrl - websocket URL
-     */
-    private async createDeltaConnection(
-        tenantId: string,
-        documentId: string,
-        token: string | null,
-        io: typeof SocketIOClientStatic,
-        client: IClient,
-        webSocketUrl: string,
-    ): Promise<OdspDocumentDeltaConnection> {
-        const startTime = performance.now();
-        const connection = await OdspDocumentDeltaConnection.create(
-            tenantId,
-            documentId,
-            token,
-            io,
-            client,
-            webSocketUrl,
-            this.mc.logger,
-            60000,
+        const module = await import(/* webpackChunkName: "internalModule" */"./internalModule");
+        this.odspDocumentServiceDelayLoaded = new module.OdspDocumentServiceDelayLoaded(
```

Comment:
Somehow, we need to set that promise before await above. Otherwise, you can have two concurrent calls coming into here, and going through this code twice. While it may work and be correct, I think we should avoid it.
You can leverage Deferred<> object, or just do new Promise() from here to ensure variable tracking delay loading is set once and on synchronous part of the API call

Reply:
Removed